### ISSUE TITLE
Requesting the training curves for ControlNet (loss, as well as MSE and SSIM on the validation set at different epochs)

### ISSUE THREAD

**Issue by Markaeov** [LemuelPuglisi/BrLP#13]
Requesting the training curves for ControlNet (loss, as well as MSE and SSIM on the validation set at different epochs). During my training on my own dataset, MSE and SSIM never improve, and I want to see what a normal training process looks like.

**Comment by LemuelPuglisi** [comment]
Hi, I should be able to recover the training curves in a few days. In the meantime, here are a few hints: the effect of ControlNet conditioning doesn’t appear gradually; instead, there’s a stage after which the diffusion model starts leveraging the condition to guide the generation process. This behaviour is consistent with the findings in the ControlNet paper (see Figure 4 in https://arxiv.org/pdf/2302.05543). I also strongly recommend adding visualisations of the generated brains to check whether they align with the baseline images.

**Comment by Markaeov** [comment]
Thank you for your patient answers! I’d like to know approximately at which step the “sudden convergence” occurs in this task. In addition, during training I observed that the generated brain images aligned with the baseline after only a few steps. However, when calculating SSIM, MSE, and PSNR, I noticed a strange phenomenon: the MSE and PSNR outperformed the metrics reported in the original paper (0.002, 26), but the SSIM was lower (0.74) and did not improve even with longer training. It’s possible that my validation code has an error. Would it be possible for you to share your validation code?

<img width="680" height="218" alt="Image" src="https://github.com/user-attachments/assets/537ef0bc-edff-4d19-a9a5-5b43e612d9e5" />

<img width="679" height="238" alt="Image" src="https://github.com/user-attachments/assets/6ec309a6-23b0-4650-a62e-5fa1af85c12e" />

**Comment by LemuelPuglisi** [comment]
The result looks great!

For the validation code, please email me at lemuel.puglisi@phd.unict.it, and I’ll provide you with a standalone evaluation script. I’ll also add the script to this repo once I have some free time.

**Comment by LemuelPuglisi** [comment]
Addressed via email.

**Comment by AlexNmSED** [comment]
Hello, I also need the validation code, and I have sent an email to you.
